gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PlanServiceImpl.java

```diff
@@ -727,6 +727,10 @@ private void assertPlanSecurityIsAllowed(final ExecutionContext executionContext
     @Override
     public boolean anyPlanMismatchWithApi(List<String> planIds, String apiId) {
         try {
+            Set<Plan> plans = planRepository.findByApi(apiId);
+            if (!plans.isEmpty() && planIds.isEmpty() || planIds.size() < plans.size()) {
```

Comment:
If we introduce this change we have to keep in mind that the doc needs to be updated

https://docs.gravitee.io/apim/3.x/apim_publisherguide_import_apis.html#plans